ethereum/eth/src/test/java/org/hyperledger/besu/ethereum/eth/transactions/TransactionPoolFactoryTest.java

```diff
@@ -262,33 +264,38 @@ public void incomingTransactionMessageHandlersRegisteredIfNoInitialSync() {
                 "transaction messages handler should be enabled"));
   }
 
+  @Test
+  public void txPoolStartsDisabledWhenInitialSyncPhaseIsRequired() {
+    // when using any of the initial syncs (SNAP, FAST, ...), txpool starts disabled
+    // and is enabled only after it is in sync
+    setupInitialSyncPhase(true);
+    assertThat(pool.isEnabled()).isFalse();
+  }
+
+  @Test
+  public void callingGetNextNonceForSenderOnDisabledTxPoolWorks() {
+    setupInitialSyncPhase(true);
+
+    assertThat(pool.getNextNonceForSender(Address.fromHexString("0x123abc"))).isEmpty();
+  }
+
+  @Test
+  public void txPoolStartsEnabledWhenFullSyncConfigured() {
+    // when using FULL sync, txpool starts enabled, because it could be
```

Comment:
Glad that you added this comment here, I was wondering why we enable txpool on full sync thinking of a node that is still syncing up to head when I first read the test name.